libgalois/include/katana/Range.h

```diff
@@ -32,6 +32,73 @@
 
 namespace katana {
 
+/**
+ * Finds the midpoint of a range.  The first half is always be bigger than
+ * the second half if the range has an odd length.
+ */
+template <typename Iterator>
+Iterator
+split_range(Iterator b, Iterator e) {
+  std::advance(b, (std::distance(b, e) + 1) / 2);
+  return b;
+}
+
+/**
+ * Returns a contiguous block from the range based on the number of divisions
+ * and the index of the block requested.
+ */
+template <typename Iterator>
+std::enable_if_t<
+    !std::is_integral<Iterator>::value, std::pair<Iterator, Iterator>>
+block_range(Iterator begin, Iterator end, unsigned idx, unsigned num) {
```

Comment:
minor: making these parameters consts would make me happy but may be I'm the only one who thinks like that.

Reply:
I'm personally not a fan of `const` on values, though I understand why we might want it.

Philosophically, it feels weird because it leaks an implementation detail to callers.

It also looks very similar to `const T&`, and it is a common error for someone to write `const T` when they really meant `const T&`. And on balance, I think the benefit for single-assigment semantics over just value semantics isn't that huge. Though, I would agree for references or pointers having the single-assignment property is very useful.

Reply:
OK. I'm not insisting on it. I just wish parameters would be const unless specified otherwise, but that's not how C++ is.